mono/arch/arm64/arm64-codegen.h

```diff
@@ -915,4 +915,118 @@ arm_encode_arith_imm (int imm, guint32 *shift)
 
 #define arm_autibsp(p) arm_format_autib ((p), 0b0011, 0b111)
 
+//---
+
+#include "../x86/x86-codegen.h"
+
+#define arm64_codegen_pre(inst)
+#define arm64_codegen_post(inst)
+
+#define ARM64_REX(bits) ((unsigned char)(0x40 | (bits)))
+
+typedef union {
+	guint64 val;
+	unsigned char b[8];
+} arm64_imm_buf;
+
+#define x86_imm_emit64(inst,imm)     \
+	do {	\
+			arm64_imm_buf imb; 	\
+			imb.val = (guint64) (imm);	\
+			*(inst)++ = imb.b [0];	\
+			*(inst)++ = imb.b [1];	\
+			*(inst)++ = imb.b [2];	\
+			*(inst)++ = imb.b [3];	\
+			*(inst)++ = imb.b [4];	\
+			*(inst)++ = imb.b [5];	\
+			*(inst)++ = imb.b [6];	\
+			*(inst)++ = imb.b [7];	\
```

Comment:
This looks wrong to me... I don't think you can emit constants on arm64 into instructions like. What does the resulting disassembly look like?